wordpress-mobile/WordPress-iOS
# Add Domains Tracks events

This PR adds Tracks events to Domains. It adds the following events:

* `domainsDashboardViewed` – visiting the Domains section from My Site
* `domainsDashboardAddDomainTapped` – tapping the button in the Domains section to choose a new domain, including whether it's using a credit or not
* `domainsSearchSelectDomainTapped` – tapping the button in Domains search to proceed with a selected domain
* `domainsRegistrationFormViewed` – when the native registration details form is displayed
* `domainsRegistrationFormSubmitted` – when the native registration details form is submitted
* `domainsPurchaseWebviewViewed` – when the webview checkout is displayed

It also restores the tracking of `.automatedTransferCustomDomainPurchased`, which was previously accidentally removed.

**To test**

* Build and run
* Navigate through the various events listed above and ensure they're logged in your console
* Here are two example runs:

```
🔵 Tracked: domains_dashboard_viewed <blog_id: 123456789, site_type: blog, tap_source: row>
🔵 Tracked: domains_dashboard_add_domain_tapped <blog_id: 123456789, site_type: blog, using_credit: false>
🔵 Tracked: automated_transfer_custom_domain_suggestion_queried <>
🔵 Tracked: automated_transfer_custom_domain_suggestion_selected <>
🔵 Tracked: domains_dashboard_add_domain_tapped <blog_id: 123456789, site_type: blog, using_credit: false>
🔵 Tracked: domains_purchase_webview_viewed <blog_id: 123456789, site_type: blog, using_credit: false>
```

```
🔵 Tracked: domain_credit_redemption_tapped <>
🔵 Tracked: automated_transfer_custom_domain_suggestion_queried <>
🔵 Tracked: automated_transfer_custom_domain_suggestion_selected <>
🔵 Tracked: domains_dashboard_add_domain_tapped <blog_id: 123456789, site_type: blog, using_credit: true>
🔵 Tracked: domains_registration_form_viewed <using_credit: true>
🔵 Tracked: domains_registration_form_submitted <using_credit: true>
🔵 Tracked: automated_transfer_custom_domain_contact_information_validated <>
🔵 Tracked: automated_transfer_custom_domain_purchased <>
🔵 Tracked: domain_credit_redemption_success <>
```

## Regression Notes

1. Potential unintended areas of impact

None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
3. What automated tests I added (or what prevented me from doing so)

N/A

PR submission checklist:
